torch/distributed/_shard/checkpoint/metadata.py

```diff
@@ -5,52 +5,39 @@
 import torch
 from torch.distributed._shard.sharded_tensor import (
     ShardedTensor,
-    ShardedTensorMetadata,
-    ShardMetadata,
 )
+from torch.distributed._shard.sharded_tensor.metadata import TensorProperties
 
-TENSOR_TYPE = Union[torch.Tensor, ShardedTensor]
-STATE_DICT_TYPE = Dict[str, Any]
-
-@dataclass
-class ShardStorageMetadata:
-    shard_metadata: ShardMetadata
-    # storage key used for this particular Shard
-    storage_key: str
-
-
-# Metadata for each param.
 @dataclass
-class ShardedTensorStorageMetadata:
-    # Metadata for the sharded tensor itself
-    tensor_metadata: ShardedTensorMetadata
-
-    # Storage info for each Shard. There's no ordering requirement for this list.
-    storage_metadata: List[ShardStorageMetadata]
+class ChunkStorageMetadata:
+    """
+    Each chunk is expected to have the same properties of the TensorStorageMetadata that includes it.
+    """
+    offsets: torch.Size
+    sizes: torch.Size
 
+    # This is reported by the storage layer so it might include serialization overhead
+    size_in_bytes: int
```

Comment:
could this be inferred by calculating with tensor properties? i.e. sizes() * dtype size? Also do we need to record `size_in_bytes` for every chunk or we can keep it for a single tensor?

Reply:
Yeah, the vast majority of the size can be calculated from `sizes() * dtype` but it would ignore non tensor data overhead.

TBH, I'm not sure I want to keep this field as I haven't found an use for it beyond being informative and the same  information is available on the storage metadata.

Reply:
On whether it could be calculated for the whole tensor. It's definitely possible, but we'd have the same issue as discussed here.

Reply:
We discussed it offline and the field will be removed until we find an use for it.